net/http/client.cpp

```diff
@@ -104,11 +111,19 @@ enum RoundtripStatus {
 
 class ClientImpl : public Client {
 public:
-    PooledDialer m_dialer;
+    PooledDialer *m_dialer;
     CommonHeaders<> m_common_headers;
     ICookieJar *m_cookie_jar;
     ClientImpl(ICookieJar *cookie_jar) :
-        m_cookie_jar(cookie_jar) {}
+        m_cookie_jar(cookie_jar),
+        m_dialer(new PooledDialer()) {}
+
+    ClientImpl(ICookieJar *cookie_jar, TLSContext *tls_ctx) :
+        m_cookie_jar(cookie_jar),
+        m_dialer(new PooledDialer(tls_ctx)) {
```

Comment:
m_dialer(tls_ctx)?

So there will be no need to change m_dialer to a pointer.

Reply:
done